.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
+---------------------------------
 
-Please provide issue report in the format that we request, EACH DETAIL IS A HUGE HELP.
+**This guide serves to set clear expectations for everyone involved with the project so that
+we can improve it together while also creating a welcoming space for everyone to participate.
+Following these guidelines will help ensure a positive experience for contributors and
+maintainers alike.**
 
-Issues that are not following the guidelines,
-will be processed as last priority or never or simply closed as invalid.
+:octocat: *Thanks for your interest in contributing to CheckStyle!* :octocat:
 
-## Contributing Guide
+### Contents
+- [Getting Started](#rocket-getting-started)
+- [Code of Conduct](#book-code-of-conduct)
+- [Asking Questions](#bulb-asking-questions)
+- [Opening an Issue](#inbox_tray-opening-an-issue)
+- [Reporting Security Issues](#lock-reporting-security-issues)
+- [Submitting Pull Requests](#technologist-submitting-pull-requests)
+- [Code Review](#white_check_mark-code-review)
+- [Google Summer of Code (GSoC)]()
+- [Credits](#coin-credits)
 
-Please note that PRs are looked only for approved issues. Follow the PR guide for opening a PR
+## :rocket: Getting Started
 
-* [Pull Request Template](https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
+ - Please see the [CheckStyle Documentation](https://checkstyle.org/beginning_development.html)
+    for information on how to get started with the project. This includes setting up your
+    development environment, building the project, and running tests.
+ - Take a look at the [Contribution Guidelines](https://checkstyle.org/contributing.html) for
+    on how to contribute to the project.
+ - Select an issue to work on from the
+    [Issues Page](https://github.com/checkstyle/checkstyle/issues). We
+    have a few issues labeled as
+    https://github.com/checkstyle/checkstyle/labels/good%20first%20issue to
+    help you get started. Once you get your first PR merged, you can move on to
+    https://github.com/checkstyle/checkstyle/labels/good%20second%20issue,
+    https://github.com/checkstyle/checkstyle/labels/good%20third%20issue,
+    and finally https://github.com/checkstyle/checkstyle/labels/good%20fourth%20issue
+    before taking up some bug fixes or features.
 
-Please follow the below links in order to start contributing to the checkstyle project
+## :book: Code of Conduct
 
-* [Beginning Development](https://checkstyle.org/beginning_development.html)
+This project and everyone participating in it is governed by the
+    [CheckStyle Code of Conduct](/.github/CODE_OF_CONDUCT.md).
 
-* [Contribution Guidelines](https://checkstyle.org/contributing.html)
+## :bulb: Asking Questions
 
-Find the links below to import the checkstyle project to an IDE
+See our [Discussions Page](https://github.com/checkstyle/checkstyle/discussions). In short, GitHub
```

Comment:
What about https://groups.google.com/g/checkstyle-devel ? I see some people still use it.

Reply:
added